tests/integration_tests/functional/test_aarch64_armshaker.py

```diff
@@ -0,0 +1,118 @@
+
+# pylint: disable=too-many-lines
+
+import platform
+import time
+
+import pytest
+
+from framework import utils
+from framework.utils_cpu_templates import nonci_on_arm
+
+PLATFORM = platform.machine()
+
+TIMEOUT = 4 * 60 * 60
+
+
+def test_armshaker(vm):
+    """
+    Run armshaker inside uVM
+    """
+
+    # Git clone armshaker
+    # We are using fork of `armsharker` because original version
+    # does not compile with new versions of `glibc`
+    # (`SIGSTKSZ` is not a constant anymore)
+    cmd = "git clone https://github.com/ShadowCurse/armshaker"
+    utils.run_cmd(cmd)
+
+    # Compile armshaker
+    cmd = "cd armshaker && make && cd .."
+    utils.run_cmd(cmd)
+
+    # Copy armshaker to uVM
+    vm.ssh.scp_put("armshaker/fuzzer", "/tmp/fuzzer")
+
+    # Start armshaker inside tmux in order to let it run headless
+    code, stdout, stderr = vm.ssh.execute_command(
+        "tmux new-session -d 'cd /tmp && ./fuzzer'"
+    )
+    print("tmux armshaker code: ", code)
+    print("tmux armshaker stdout: ", stdout)
+    print("tmux armshaker stderr: ", stderr)
+
+    curr_inst = ""
+    # Armshaker tests all instructions from 0 to ffffffff
+    while curr_inst != "ffffffff":
+        time.sleep(10 * 60)
+        code, stdout, stderr = vm.ssh.execute_command("cat /tmp/data/status")
+        if stdout != "":
+            curr_inst = stdout.split("\n")[0].split(":")[1]
+            print("instuctions tested: ", curr_inst)
+        else:
+            print("couldn't read /tmp/data/status")
+            print("code: ", code)
+            print("stdout: ", stdout)
+            print("stderr: ", stderr)
+
+    # Execution succeeded. Print last status
+    _, stdout, _ = vm.ssh.execute_command("cat /tmp/data/status")
+    print("final status: ", stdout)
+
+
+@pytest.mark.skipif(
+    PLATFORM != "aarch64",
+    reason="This is aarch64 specific test.",
+)
+@pytest.mark.timeout(TIMEOUT)
+@pytest.mark.nonci
+def test_armshaker_default(test_microvm_with_api):
+    """
+    Run armshaker inside default uVM
+    """
+    vm = test_microvm_with_api
+    vm.spawn()
+    vm.basic_config()
+    vm.add_net_iface()
+    vm.start()
+    test_armshaker(vm)
+
+
+@pytest.mark.skipif(
+    PLATFORM != "aarch64",
+    reason="This is aarch64 specific test.",
+)
+@pytest.mark.timeout(TIMEOUT)
+@nonci_on_arm
+def test_armshaker_with_static_template(test_microvm_with_api, cpu_template):
+    """
+    Run armshaker inside uVM with static templates
+    """
+    vm = test_microvm_with_api
+    vm.spawn()
+    vm.basic_config(cpu_template=cpu_template)
+    vm.add_net_iface()
+    vm.start()
+    test_armshaker(vm)
+
+
+@pytest.mark.skipif(
+    PLATFORM != "aarch64",
+    reason="This is aarch64 specific test.",
+)
+@pytest.mark.timeout(TIMEOUT)
+@nonci_on_arm
+def test_armshaker_with_custom_template(test_microvm_with_api, custom_cpu_template):
+    """
+    Run armshaker inside uVM with custom templates
+    """
+    vm = test_microvm_with_api
```

Comment:
nit: would it have been better if `microvm_factory.build` was used instead of `test_microvm_with_api`? @pb8o can correct me but I think `test_microvm_with_api` is going to be removed later on so not using now it will save us some time later.